app/post.go

```diff
@@ -586,6 +596,45 @@ func (a *App) GetPostsAroundPost(postId, channelId string, offset, limit int, be
 	}
 }
 
+func (a *App) GetPostsForChannelAroundLastUnread(channelId, userId string) (*model.PostList, *model.AppError) {
+	var member *model.ChannelMember
+	var err *model.AppError
+	if member, err = a.GetChannelMember(channelId, userId); err != nil {
+		return nil, err
+	}
+
+	var postListSince *model.PostList
+	if member.LastViewedAt > 0 {
```

Comment:
What happens if the user has never viewed the channel before (ie. `member.LastViewedAt` is 0)? Since that's an edge case, maybe we should do an early exit from the function to simplify the later logic

Reply:
Also, is there a test for this case?

Reply:
👍 and also added the test for this case.